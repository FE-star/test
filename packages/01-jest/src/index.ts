export function add(a: number, b: number): number {
  return a + b;
}

export function addFloat(a: number, b: number): number {
  const scale = 1000000;
  return (a * scale + b * scale) / scale;
}
