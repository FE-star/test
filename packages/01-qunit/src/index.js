function add(a, b) {
  return a + b;
}

function addFloat(a, b) {
  const scale = 1000000;
  return (a * scale + b * scale) / scale;
}

QUnit.test('test add()', function (assert) {
  assert.equal(add(1, 2), 3, 'add(1, 2) = 3');
});

QUnit.test('test addFloat()', function (assert) {
  assert.equal(addFloat(0.1, 0.2), 0.3, 'addFloat(0.1, 0.2) = 0.3');
});

// QUnit.test('test add() for float', function (assert) {
//   assert.equal(add(.1, .2), .3, 'add(.1, .2) = .3');
// });
