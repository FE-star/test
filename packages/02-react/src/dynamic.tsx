import React from 'react';
import { createRoot } from 'react-dom/client';

export function createDialog(props: { title: string; content: string }) {
  const { title, content } = props;
  const div = document.createElement('div');
  document.body.appendChild(div);

  const root = createRoot(div);

  setTimeout(() => {
    root.render(
      <div className="dialog">
        <div className="dialog-title">{title}</div>
        <div className="dialog-content">{content}</div>
      </div>
    );
  }, 200)
}
