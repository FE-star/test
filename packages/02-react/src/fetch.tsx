import React, { useEffect, useState, act } from 'react';

type DataItem = {
  id: string;
  name: string;
};

type DataResult = {
  dataList: DataItem[];
  success: boolean;
};

export function App() {
  const [list, setList] = useState<DataItem[]>([]);

  useEffect(() => {
    window
      .fetch('https://example.com/getData')
      .then((res) => {
        return res.json();
      })
      .then((result: DataResult) => {
        if (result.success === true) {
          act(() => {
            setList(result.dataList);
          });
        }
      })
      .catch((err) => {
        console.log('window.fetch 获取数据失败，错误信息', err);
      });
  }, []);

  return (
    <div className="app">
      <div className="data-list">
        {list.map((item) => {
          return (
            <div className="data-item" key={item.id}>
              {item.name}
            </div>
          );
        })}
      </div>
    </div>
  );
}
