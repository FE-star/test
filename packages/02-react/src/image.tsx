import React, { useState, act } from 'react';

const image404 = 'https://example/404.png';

export function App(props: { src: string; onLoad?: () => void; onError?: () => void }) {
  const { src, onLoad, onError } = props;

  const [imageSrc, setImageSrc] = useState<string>(src);

  return (
    <div className="app">
      <img
        src={imageSrc}
        onLoad={() => {
          onLoad?.();
        }}
        onError={() => {
          act(() => {
            setImageSrc(image404);
          });
          onError?.();
        }}
      />
    </div>
  );
}
