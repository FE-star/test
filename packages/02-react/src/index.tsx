import React, { useState } from 'react';

export function App() {
  const [count, setCount] = useState(0);

  return (
    <div className="app-counter">
      <div className="counter-text">count:{count}</div>
      <button
        className="counter-btn"
        onClick={() => {
          setCount(count + 1);
        }}
      >
        Add
      </button>
    </div>
  );
}
