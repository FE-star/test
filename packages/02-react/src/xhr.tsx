import React, { useEffect, useState, act } from 'react';

type DataItem = {
  id: string;
  name: string;
};

type DataResult = {
  dataList: DataItem[];
  success: boolean;
};

export function App() {
  const [list, setList] = useState<DataItem[]>([]);

  useEffect(() => {
    var xhr = new XMLHttpRequest();
    xhr.addEventListener('load', () => {
      const result: DataResult = JSON.parse(xhr.responseText);
      if (result.success === true) {
        act(() => {
          setList(result.dataList);
        });
      }
    });
    xhr.addEventListener('error', (err) => {
      console.log('XHR 获取数据失败，错误信息', err);
    });
    xhr.open('GET', 'https://example.com/requestData');
    xhr.send();
  }, []);

  return (
    <div className="app">
      <div className="data-list">
        {list.map((item) => {
          return (
            <div className="data-item" key={item.id}>
              {item.name}
            </div>
          );
        })}
      </div>
    </div>
  );
}
