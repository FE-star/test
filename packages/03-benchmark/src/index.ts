export function addFloat(a: number, b: number): number {
  const scale = 1000000;
  return (a * scale + b * scale) / scale;
}

export function addFloatBySplitStr(a: number, b: number): number {
  const aStrList = a.toString().split('.');
  const bStrList = b.toString().split('.');
  const resultInt = parseInt(aStrList[0]) + parseInt(bStrList[0]);
  const resultFloat = parseInt(aStrList[1]) + parseInt(bStrList[1]);
  return parseFloat(`${resultInt}.${resultFloat}`);
}
