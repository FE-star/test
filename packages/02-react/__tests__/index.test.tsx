import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { App } from '../src/index';

describe('App', () => {
  test('snapshot', () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });

  test('click', () => {
    const { container } = render(<App />);
    const text = container.querySelector('.counter-text');
    expect(text?.textContent).toBe('count:0');

    const btn = container.querySelector('.counter-btn');
    // 可以替换成 btn.click() 吗？
    // 为什么？
    fireEvent.click(btn as HTMLElement);
    expect(text?.textContent).toBe('count:1');

    fireEvent.click(btn as HTMLElement);
    expect(text?.textContent).toBe('count:2');
  });
});
