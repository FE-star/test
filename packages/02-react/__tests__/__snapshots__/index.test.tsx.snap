// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot 1`] = `
<div>
  <div
    class="app-counter"
  >
    <div
      class="counter-text"
    >
      count:
      0
    </div>
    <button
      class="counter-btn"
    >
      Add
    </button>
  </div>
</div>
`;
